tensorflow/compiler/jit/xla_launch_util.cc

```diff
@@ -254,7 +254,8 @@ XlaComputationLaunchContext::PopulateInputs(
         is_resource_variable &&
         absl::c_any_of(compilation_result->resource_updates,
                        [&](const XlaCompiler::ResourceUpdate& update) {
-                         return update.input_index == i && update.modified;
```

Comment:
Could you clarify why `i` was changed to `arg_num`?

Reply:
Because `resource_updates` stores `arg_num` (see [here](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/compiler/tf2xla/xla_compiler.cc;l=302?q=file:tensorflow%2Fcompiler%2Ftf2xla%2Fxla_compiler.cc)), which is an input index seen at the compilation time.

`i` refers to the input shapes of the compiled kernel. `arg_num` and `i` can differ in the auto-clustering mode. They might be no different for manual clustering.

Reply:
> They might be no different for manual clustering

Can we CHECK it then?